index.js

```diff
@@ -10,9 +10,85 @@ const COLORS = 'https://nt-cdn.s3.amazonaws.com/colors.json';
  * @param compHex filter for complementary color hex code
  * @returns Promise
  */
-const fetchColors = ({ name, hex, compName, compHex }) => {
-  throw Error('Not implemented');
+
+
+const fetchColors = ({ name, hex, compName, compHex }) => 
+  
+  {
+   return fetch(COLORS)
+
+    .then((response) => {
+      if (!response.ok) {
+       throw new Error(`HTTP error, status = ${response.status}`);
+      }
+      return response.json();
+    })
+
+    .then((data) => {
+
+      //create an empty array for appending future results, use "let" because it's block-scoped
+      let result = []
+
+      //iterate through data by each element
+      data.forEach(element => {
```

Comment:
a better way of doing this would be tu use `filter` instead of `forEach`. You wouldn't need to create a new array but simply filter the one passed in and return it.